faust/stores/memory.py

```diff
@@ -31,7 +31,9 @@ def apply_changelog_batch(
             self._create_batch_iterator(to_delete.add, to_key, to_value, batch)
         )
         for key in to_delete:
-            delete_key(key, None)
+            # If the key was assigned a value again at a later time, it will not be deleted.
+            if not self.data[key]:
```

Comment:
Is this an issue only for memory store? Rocksdb store does not have this issue AFAIK

Reply:
The issue is only for memory. When I look into the code for Rocksdb I don't recongnize that there can be the same error